cmd/cloud_sql_proxy/proxy_test.go

```diff
@@ -42,58 +42,137 @@ func TestCreateInstanceConfigs(t *testing.T) {
-		}, {
-			"setting -dir and -instances_metadata",
-			"dir", false, nil, "md", false, false, false,
-		}, {
-			"setting -instances_metadata",
-			"", false, nil, "md", true, false, true,
+			desc:                     "setting -dir and -instances (unix socket) w/ something invalid",
+			dir:                      "dir",
+			useFuse:                  false,
+			instances:                []string{"proj:reg:x", "INVALID_PROJECT_STRING"},
+			instancesSrc:             "",
+			wantErr:                  false,
+			skipFailedInstanceConfig: true,
+		},
+		{
+			desc:                     "Seting -instance (unix socket)",
+			dir:                      "",
+			useFuse:                  false,
+			instances:                []string{"proj:reg:x"},
+			instancesSrc:             "",
+			wantErr:                  true,
+			skipFailedInstanceConfig: false,
+		},
+		{
+			desc:                     "setting -instance (tcp socket)",
+			dir:                      "",
+			useFuse:                  false,
+			instances:                []string{"proj:reg:x=tcp:1234"},
+			instancesSrc:             "",
+			wantErr:                  false,
+			skipFailedInstanceConfig: false,
+		},
+		{
+			desc:                     "setting -instance (tcp socket) and -instances_metadata",
+			dir:                      "",
+			useFuse:                  false,
+			instances:                []string{"proj:reg:x=tcp:1234"},
+			instancesSrc:             "md",
+			wantErr:                  true,
+			skipFailedInstanceConfig: false,
+		},
+		{
+			desc:                     "setting -dir, -instance (tcp socket), and -instances_metadata",
+			dir:                      "dir",
+			useFuse:                  false,
+			instances:                []string{"proj:reg:x=tcp:1234"},
+			instancesSrc:             "md",
+			wantErr:                  false,
+			skipFailedInstanceConfig: false,
+		},
+		{
+			desc:                     "setting -dir, -instance (unix socket), and -instances_metadata",
+			dir:                      "dir",
+			useFuse:                  false,
+			instances:                []string{"proj:reg:x"},
+			instancesSrc:             "md",
+			wantErr:                  false,
+			skipFailedInstanceConfig: false,
+		},
+		{
+			desc:                     "setting -dir and -instances_metadata",
+			dir:                      "dir",
+			useFuse:                  false,
+			instances:                nil,
+			instancesSrc:             "md",
+			wantErr:                  false,
+			skipFailedInstanceConfig: false,
+		},
+		{
+			desc:                     "setting -instances_metadata",
+			dir:                      "",
+			useFuse:                  false,
+			instances:                nil,
+			instancesSrc:             "md",
+			wantErr:                  true,
+			skipFailedInstanceConfig: false,
 		},
 	} {
-		if runtime.GOOS == "windows" && !v.supportedOnWindows {
+		// fuse is not supported in CI for darwin
+		if runtime.GOOS == "darwin" && v.useFuse {
```

Comment:
nit: Would it be better to use t.Run and t.Skip here?

Reply:
I'll send a PR to clean this up.